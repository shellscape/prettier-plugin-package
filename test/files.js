/* eslint-disable import/no-unresolved, sort-keys */

const test = require('ava');
const prettier = require('prettier');

test('default', (t) => {
  const options = {
    filepath: 'package.json',
    parser: 'json-stringify',
    plugins: ['.']
  };
  const fixture = {
    files: ['README.md', 'LICENSE', 'lib/']
  };

  const input = JSON.stringify(fixture, null, 2);
  const output = prettier.format(input, options);

  t.snapshot(output);
});

test('negations', (t) => {
  const options = {
    filepath: 'package.json',
    parser: 'json-stringify',
    plugins: ['.']
  };
  const fixture = {
    files: [
      '/lit-html.js',
      '/lit-html.js.map',
      '/lit-html.d.ts',
      '/lit-html.d.ts.map',
      '/directives/',
      '/parts.js',
      '/parts.js.map',
      '/parts.d.ts',
      '/parts.d.ts.map',
      '/src/',
      '!/src/test/',
      '/development/',
      '!/development/test/'
    ]
  };

  const input = JSON.stringify(fixture, null, 2);
  const output = prettier.format(input, options);

  t.snapshot(output);
});
