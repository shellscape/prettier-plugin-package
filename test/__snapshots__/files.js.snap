// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`default 1`] = `
"{
  "files": [
    "lib/",
    "README.md",
    "LICENSE"
  ]
}
"
`;

exports[`negations 1`] = `
"{
  "files": [
    "/development/",
    "!/development/test/",
    "/directives/",
    "/lit-html.d.ts",
    "/lit-html.d.ts.map",
    "/lit-html.js",
    "/lit-html.js.map",
    "/parts.d.ts",
    "/parts.d.ts.map",
    "/parts.js",
    "/parts.js.map",
    "/src/",
    "!/src/test/"
  ]
}
"
`;
